examples/jacobi.cxx

```diff
@@ -0,0 +1,333 @@
+  circuit law. 
+
+  The structured N x N circuit may be visualized as follows: 
+
+  - Dash lines correspond to lossless wires,
+  - The circles correspond to lossless juntions
+  - The [] correspond to 1 Ohm resistors
+  - The [x] corresponds to a 1 volt DC battery which powers the grid
+  
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  [x]    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+
+
+  ----[RAJA Concepts]---------------
+  1. RAJA Reduction
+  2. RAJA::omp_collapse_nowait_exec
+
+*/
+
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout<<"Example 3: Jacobi Method for Linear System"<<std::endl;
+
+  //----[Parameters for Solver]-------
+  double tol = 1e-5;
+  int maxIter = 10000;
+  int N = 100;
+  int NN=(N+2)*(N+2);
+  int iteration;
+  double resI2, V, invD;
+
+  double *I    = allocate<double>(NN);
+  double *Iold = allocate<double>(NN);
+
+  //Set to zero
+  memset(I,0,NN*sizeof(double));
+  memset(Iold,0,NN*sizeof(double));
+
+
+  //----[Standard C++ Loop]------
+  printf("Standard  C++ Loop \n");
+  resI2 = 1;  iteration = 0; 
+
+  //Carry out iterations of the Jacobi scheme until a tolerance is met
+  while(resI2>tol*tol) {
+    
+    //Iteration of the Jacobi Scheme
+    for(int n=1;n<=N;++n) {
+      for(int m=1;m<=N;++m) {
+
+        int id = n*(N+2) + m;
+
+        //Cell (1,1) is a special case due to battery
+        if(n==1 && m==1) {
+          invD = 1./3.0; V = 1.0; 
+          I[id] = invD*(V-Iold[id-N-2]-Iold[id+N+2]-Iold[id-1]-Iold[id+1]);
+        }else{
+          invD = 1./4.0; V = 0.0;    
+          I[id] = invD*(V-Iold[id-N-2]-Iold[id+N+2]-Iold[id-1]-Iold[id+1]);
+        }
+        
+      }
+    }
+
+    //Compute ||I - I_{old}||_{l2} and update Iold
+    resI2 = 0.0; 
+    for(int k=0; k<NN; k++) {
+      resI2 += (I[k]-Iold[k])*(I[k]-Iold[k]);
+      Iold[k]=I[k];
+    }
+    
+    if(iteration > maxIter) {        
+      printf("Standard C++ Loop - Maxed out on iterations \n");
+      exit(-1);
+    }
+
+    iteration++;
+  }
+  printf("Top right current: %lg \n", I[N+N*(N+2)]);
+  printf("No of iterations: %d \n \n",iteration);
+  //======================================
+
+  //----[RAJA: Sequential Policy - Nested forallN]---------
+  printf("RAJA: Sequential Policy - Nested forallN \n"); 
+  resI2 = 1; iteration = 0; 
+  memset(I,0,NN*sizeof(double));
+  memset(Iold,0,NN*sizeof(double));
+  while(resI2 > tol*tol) {
```

Comment:
For the RAJA cases in this example, I think it would be worth making the point that one can reuse RAJA code for different backing models here.  A function with the exec and reduce policies as parameters would be all it would take to abstract that out and remove the replication. (well, maybe also a string or two, but close enough)